### PR TITLE
Implements screenshot feature for dell

### PR DESCRIPTION
**Pull request by joelrebel**
## What does this PR implement/change/remove?

Implements screenshot capture feature for Dell IDracs.

### Checklist
- [x] Tests added
- [X] Similar commits squashed

### The HW vendor this change applies to (if applicable)

Dell

## Description for changelog/release notes

```
- Screenshot feature for Dell Idracs
```

### CODE REVIEW

**Comment by joelrebel** [comment]
rebase on main with changes

File: bmc/screenshot.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+			metadataLocal.SuccessfulProvider = elem.name
+			return image, fileType, metadataLocal, nil
+		}
+	}
+
+	return image, fileType, metadataLocal, multierror.Append(err, errors.New("failed to capture screenshot"))
+}
+
+// ScreenshotFromInterfaces identifies implementations of the ScreenshotGetter interface and passes the found implementations to the screenshot() wrapper method.
+func ScreenshotFromInterfaces(ctx context.Context, generic []interface{}) (image []byte, fileType string, metadata Metadata, err error) {
```
Comment (chrisdoherty4) [inline bmc/screenshot.go:52]: What needs this signature?

Comment (joelrebel) [reply bmc/screenshot.go:52]: Its how the client passes in the provider registry here - https://github.com/bmc-toolbox/bmclib/blob/49f837d37ccca73f85ca223d214988f36d2417f5/client.go#L364

**Review by chrisdoherty4 — APPROVED** [review]
lgtm

**Comment by jacobweinstock** [comment]
Hey @joelrebel , a lot of this looks like duplicate code of the redfish provider. Seems like there are just some specific dell endpoints for being able to screenshot. What would this look like if it was incorporated into the redfish provider? Would it make it too messy?

**Comment by joelrebel** [comment]
@jacobweinstock I did take that approach earlier, and the Redfish provider starts to get [messy](https://github.com/bmc-toolbox/bmclib/blob/main/providers/redfish/tasks.go#L29), which now I'd like to move into the relevant provider. In this case we need to support a minimal set of features that is vendor specific - not present in Redfish spec (and most likely will never be), and so it made more sense have a provider implementation for it.

**Comment by jacobweinstock** [comment]
> @jacobweinstock I did take that approach earlier, and the Redfish provider starts to get [messy](https://github.com/bmc-toolbox/bmclib/blob/main/providers/redfish/tasks.go#L29), which now I'd like to move into the relevant provider. In this case we need to support a minimal set of features that is vendor specific - not present in Redfish spec (and most likely will never be), and so it made more sense have a provider implementation for it.

Ah i see. Thanks!

**Pull request merged**